pkg/forwarder/forwarder.go

```diff
@@ -184,6 +184,7 @@ type Options struct {
 	APIKeyValidationInterval time.Duration
 	KeysPerDomain            map[string][]string
 	ConnectionResetInterval  time.Duration
+	CompletionHandler        HTTPCompletionHandler
```

Comment:
Where does this get used? I mean I see it getting set, but what is it for?

Reply:
we're hooking the `apiWatcher` into this. that allows us to detect if we're not being able to submit the metrics "directly" via the API. if that's the case we flush the metrics via statsd (and thus forward it to the agent-core)